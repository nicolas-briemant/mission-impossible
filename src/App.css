@import '../node_modules/@blueprintjs/core/dist/blueprint.css';

body {
  width: 1000px;
  margin: auto;
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.App-img {
  position: absolute;
  height: 66px;
  width: auto;
}

.red {
  color: red;
}

.green {
  color: green;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
