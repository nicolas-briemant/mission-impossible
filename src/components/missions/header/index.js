import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { ButtonGroup, Button } from '@blueprintjs/core';
import { Header, HeaderLeft, HeaderRight } from './header';
import logo from '../../../data/logo.png';

const Logo = glamorous.img({
  width: 75,
});

const Header2 = ({ blockAction }) => (
  <Header>
    <HeaderLeft>
      <Logo alt="Logo" src={logo} />
      <h4>Titre du site</h4>
    </HeaderLeft>
    <HeaderRight>
      <ButtonGroup minimal="true" large="true">
        <Button iconName="pt-icon-align-justify" onClick={() => blockAction()}>
          Missions
        </Button>
        <Button iconName="people" onClick={() => blockAction()}>
          Peoples
        </Button>
      </ButtonGroup>
    </HeaderRight>
  </Header>
);

Header2.propTypes = {
  blockAction: PropTypes.func,
};

export default Header2;
