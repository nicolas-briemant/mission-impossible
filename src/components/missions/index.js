import React, { Component } from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import cx from 'classnames';
import { Button } from '@blueprintjs/core';
import 'normalize.css/normalize.css';
import '@blueprintjs/core/dist/blueprint.css';
import logger from '../logger';

const StyleMission = glamorous.div(
  {
    width: '300px',
    padding: '20px',
    textAlign: 'center',
    border: '1px solid #c8c8c8',
  },
  ({ isSelected }) => ({
    backgroundColor: isSelected ? 'indianred' : 'white',
  }),
);

const MainContainer = glamorous.div({
  margin: '2rem',
  padding: '3rem',
  border: '1px solid gray',
});

const Toolbar = glamorous.div({
  padding: '1rem',
  fontSize: '1.5em',
  border: '1px solid gray',
  textAlign: 'left',
  marginBottom: '1em',
  display: 'flex',
  justifyContent: 'space-between',
  borderRadius: '3px',
  alignItems: 'center',
});

const MissionsContainer = glamorous.div({
  display: 'flex',
  flexWrap: 'wrap',
  justifyContent: 'center',
});

class Mission extends Component {
  constructor(props) {
    super(props);
    this.state = { isHovered: false };
    this.boundUpdateIsHovered = this.updateIsHovered.bind(this);
  }

  updateIsHovered() {
    this.setState({ isHovered: !this.state.isHovered });
  }

  render() {
    const { id, name, clientId, partnerId, managerId, addenda, isSelected, removeMission, toggleMission } = this.props;
    return (
      <StyleMission
        isHovered={this.state.isHovered}
        isSelected={isSelected}
        onMouseEnter={() => this.boundUpdateIsHovered()}
        onMouseLeave={() => this.boundUpdateIsHovered()}
      >
        <h3>{name}</h3>
        <p>
          clientId: {clientId}, partnerId: {partnerId}, managerId: {managerId}
        </p>
        <p>
          Workers ({addenda.length}): {addenda.map(w => w.workerId).join(' ')}
        </p>
        {this.state.isHovered ? (
          <div>
            <Button iconName="trash" text="Supprimer" onClick={() => removeMission(id)} />
            <Button iconName="add" text="SÃ©lectionner" onClick={() => toggleMission(id)} />
          </div>
        ) : null}
      </StyleMission>
    );
  }
}

Mission.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  clientId: PropTypes.string,
  partnerId: PropTypes.string,
  managerId: PropTypes.string,
  addenda: PropTypes.arrayOf(
    PropTypes.shape({
      workerId: PropTypes.string.isRequired,
    }),
  ).isRequired,
  isSelected: PropTypes.bool,
  removeMission: PropTypes.func.isRequired,
  toggleMission: PropTypes.func.isRequired,
};

const Missions = ({ missions, removeMission, toggleMission, removeMissions }) => {
  const colorNbMissions = {
    red: false,
    green: true,
  };

  const isAnyoneIsSelected = missions.filter(e => e.isSelected === true);
  return (
    <MainContainer>
      <Toolbar>
        <i className={cx((colorNbMissions: missions.length))}>Nombre de mission(s) : {missions.length}</i>
        <div>
          {isAnyoneIsSelected.length ? (
            <button type="button" className="pt-button pt-intent-danger" onClick={() => removeMissions()}>
              Supprimer la selection : ({isAnyoneIsSelected.length})
              <span className="pt-icon-standard pt-icon-cross pt-align-right" />
            </button>
          ) : null}
          <button type="button" className="pt-button pt-intent-primary">
            Filtre
            <span className="pt-icon-standard pt-icon-arrow-right pt-align-right" />
          </button>
        </div>
      </Toolbar>
      <MissionsContainer>
        {missions.map(mission => (
          <Mission key={mission.id} {...mission} removeMission={removeMission} toggleMission={toggleMission} />
        ))}
      </MissionsContainer>
    </MainContainer>
  );
};

Missions.propTypes = {
  missions: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
    }),
  ).isRequired,
  removeMission: PropTypes.func.isRequired,
  toggleMission: PropTypes.func.isRequired,
  removeMissions: PropTypes.func.isRequired,
};

export default logger('Missions')(Missions);
