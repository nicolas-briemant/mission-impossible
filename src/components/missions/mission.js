import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import uniqid from 'uniqid';
import { Button } from '@blueprintjs/core';
import cx from 'classnames';
import glamorous from 'glamorous';
import moment from 'moment';

const MyUlCont = glamorous.ul(
  {
    width: '311px',
    margin: '10px',
    border: 'solid 1px black',
    padding: '0px',
  },
  ({ isSelected }) => ({
    backgroundColor: isSelected ? 'lightcoral' : 'white',
  }),
);

const Mylist = glamorous.div({
  display: 'flex',
  justifyContent: 'center',
  marginTop: '10px',
  marginBottom: '10px',
  ':hover': {
    fontWeight: 'bold',
  },
  ':active': {
    color: 'red',
    position: 'relative',
  },
});

const AlignCenter = glamorous.div({
  display: 'flex',
  justifyContent: 'center',
});

const MyDivCont = glamorous.div({
  display: 'flex',
  flexWrap: 'wrap',
});

class Mission extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isHovered: false,
    };
    this.boundUpdateIsHovered = this.updateIsHovered.bind(this);
  }

  updateIsHovered = () => {
    this.setState({
      isHovered: !this.state.isHovered,
    });
  };

  render() {
    const { mission, removeMission, selectMission, selectedMissions, count } = this.props;
    const { id, name, client, partner, addenda, manager, startDate, endDate } = mission;
    if (count <= 0) {
      const noMission = (
        <MyDivCont>
          <p> No mission </p>
        </MyDivCont>
      );
      return noMission;
    }
    const today = moment().subtract(1, 'days');
    const isWip = endDate < today;

    const cls = cx({
      'pt-tag pt-intent-sucess': isWip,
      'pt-tag pt-intent-warning': !isWip,
    });

    return (
      <Fragment>
        <MyUlCont
          isHovered={this.state.isHovered}
          isSelected={selectedMissions[mission.id]}
          onMouseEnter={this.updateIsHovered}
          onMouseLeave={this.updateIsHovered}
        >
          <Mylist>Mission: {name} </Mylist>
          <Mylist>Client: {client.name} </Mylist>
          {partner ? <Mylist>Partner: {partner.name}</Mylist> : null}
          <Mylist>
            Manager : {manager.firstName} {manager.lastName}
          </Mylist>
          {addenda.map(({ worker }) => (
            <div key={uniqid()}>
              <Mylist>
                Worker: {worker.firstName} {worker.lastName}
              </Mylist>
            </div>
          ))}
          <Mylist>{startDate.format('LL')} </Mylist>
          <Mylist>{endDate ? `End Date: ${endDate.format('LL')}` : null} </Mylist>
          <Mylist className={cls}>{isWip ? 'Termined' : 'Work in progress...'}</Mylist>
          {this.state.isHovered ? (
            <AlignCenter>
              <Button iconName="trash" text="remove" onClick={() => removeMission(id)} />
              <Button iconName="add" text="select" onClick={() => selectMission(id)} />
            </AlignCenter>
          ) : null}
        </MyUlCont>
      </Fragment>
    );
  }
}

Mission.propTypes = {
  mission: PropTypes.object.isRequired,
  removeMission: PropTypes.func.isRequired,
  selectMission: PropTypes.func.isRequired,
  selectedMissions: PropTypes.object.isRequired,
  count: PropTypes.number.isRequired,
};

export default Mission;
