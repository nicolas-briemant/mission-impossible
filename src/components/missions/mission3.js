import React from 'react';

export const Mission3 = () => (
  <div>name: mission3, client: client3</div>
);
