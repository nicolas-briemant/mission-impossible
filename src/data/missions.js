const missions = [
  {
    _id: '1',
    name: 'React Components',
    clientId: '566ac1d996de2706000c9484',
    managerId: '2',
    addenda: [
      {
        workerId: '6',
        startDate: '2015-10-31T23:00:00.000Z',
        endDate: '2016-05-31T23:00:00.000Z',
        fees: { amount: 350, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '6',
        startDate: '2016-06-01T23:00:00.000Z',
        fees: { amount: 420, unit: 'day', currency: 'EUR' },
      },

      {
        workerId: '5',
        startDate: '2015-10-31T23:00:00.000Z',
        fees: { amount: 420, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '2',
        startDate: '2014-10-31T23:00:00.000Z',
        fees: { amount: 520, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '1',
        startDate: '2017-04-11T23:00:00.000Z',
        endDate: '2017-07-11T23:00:00.000Z',
        fees: { amount: 520, unit: 'day', currency: 'EUR' },
      },
    ],
  },
  {
    _id: '2',
    name: 'Groupagora',
    clientId: '566acaba96de2706000c9488',
    managerId: '1',
    addenda: [
      {
        workerId: '1',
        startDate: '2017-04-11T23:00:00.000Z',
        fees: { amount: 520, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '3',
        startDate: '2017-03-11T23:00:00.000Z',
        fees: { amount: 350, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '4',
        startDate: '2017-03-11T23:00:00.000Z',
        endDate: '2017-15-11T23:00:00.000Z',
        fees: { amount: 0, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '4',
        startDate: '2017-16-11T23:00:00.000Z',
        fees: { amount: 370, unit: 'day', currency: 'EUR' },
      },
    ],
  },
  {
    _id: '3',
    name: 'BFI',
    clientId: '566aef4496de2706000c9494',
    partnerid: '5671444574cd4006000b911c',
    managerId: '2',
    addenda: [
      {
        workerId: '7',
        startDate: '2018-01-11T23:00:00.000Z',
        fees: { amount: 280, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '9',
        startDate: '2018-01-11T23:00:00.000Z',
        fees: { amount: 280, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '10',
        startDate: '2018-01-11T23:00:00.000Z',
        fees: { amount: 280, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '8',
        startDate: '2018-01-11T23:00:00.000Z',
        endDate: '2018-03-11T23:00:00.000Z',
        fees: { amount: 280, unit: 'day', currency: 'EUR' },
      },
      {
        workerId: '8',
        startDate: '2018-03-12T23:00:00.000Z',
        fees: { amount: 380, unit: 'day', currency: 'EUR' },
      },
    ],
  },
  {
    _id: '4',
    name: 'Graph satelites',
    clientId: '568a68c8c08f1907004483c8',
    partnerid: '568c153be355a707004f003c',
    managerId: '1',
    addenda: [
      {
        workerId: '7',
        startDate: '2017-01-11T23:00:00.000Z',
        endDate: '2018-01-11T23:00:00.000Z',
        fees: { amount: 480, unit: 'day', currency: 'EUR' },
      },
    ],
  },
];

export default missions;
