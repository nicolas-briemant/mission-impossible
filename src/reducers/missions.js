import { reject, equals, prop } from 'ramda';
import {
  REMOVE_MISSION,
  REMOVE_MISSIONS,
  TOGGLE_MISSION,
  SORT_MISSIONS,
  FILTER_MISSIONS,
  SPOTLIGHT_MISSIONS,
} from '../actions';

export default (initialState) => (state = initialState, action = {}) => {
  switch(action.type) {
    case REMOVE_MISSION:
      return {
        ...state,
        collection: reject(mission => equals(mission.id, action.payload.missionId), state.collection),
      };
    case REMOVE_MISSIONS:
      return {
        ...state,
        collection: reject(prop('isSelected'), state.collection),
      };
    case TOGGLE_MISSION:
      return {
        ...state,
        collection: {
          ...state.collection,
          [action.payload.missionId]: {
            ...state.collection[action.payload.missionId],
            isSelected: !state.collection[action.payload.missionId].isSelected,
          },
        },
      };
    case SORT_MISSIONS:
      return {
        ...state,
        sort: {
          ...state.sort,
          id: action.payload.sortId,
          direction: state.sort.id === action.payload.sortId ? !state.sort.direction : true,
        },
      };
    case FILTER_MISSIONS:
      return {
        ...state,
        filter: action.payload.filterId,
      };
    case SPOTLIGHT_MISSIONS:
      return {
        ...state,
        spotlight: action.payload.term,
      };
    default:
      return state;
  }
};
