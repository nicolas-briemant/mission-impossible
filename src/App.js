import React from 'react';
import logo from './logo.svg';
import './App.css';
import Missions from './components/missions';

// class App extends Component {
//   render() {
//     return (
//       <div className="App">
//         <header className="App-header">
//           <img src={logo} className="App-logo" alt="logo" />
//           <h1 className="App-title">Welcome to mission impossible</h1>
//         </header>
//         <Missions />
//       </div>
//     );
//   }
// }

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h1 className="App-title">Welcome to mission impossible</h1>
      </header>
      <Missions />
    </div>
  );
}
export default App;
