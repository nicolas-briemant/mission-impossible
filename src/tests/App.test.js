import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from '../App';
import missions from '../data/missions2';

const data = {
  missions,
};

describe('App', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App {...data} />, div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<App {...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
