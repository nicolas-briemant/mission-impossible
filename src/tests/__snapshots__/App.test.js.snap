// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    missions={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="IMAGE_MOCK"
          />
           
          <h1
                    className="App-title"
          >
                    Welcome to mission impossible
          </h1>
</header>,
        <Missions
          missions={Array []}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="IMAGE_MOCK"
/>,
            " ",
            <h1
              className="App-title"
>
              Welcome to mission impossible
</h1>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "IMAGE_MOCK",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to mission impossible",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Welcome to mission impossible",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "missions": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="IMAGE_MOCK"
            />
             
            <h1
                        className="App-title"
            >
                        Welcome to mission impossible
            </h1>
</header>,
          <Missions
            missions={Array []}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="IMAGE_MOCK"
/>,
              " ",
              <h1
                className="App-title"
>
                Welcome to mission impossible
</h1>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "IMAGE_MOCK",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to mission impossible",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Welcome to mission impossible",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "missions": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
